apps/backend/src/libs/hooks/check-user-permissions.hook.ts

```diff
@@ -6,27 +6,39 @@ import { type APIPreHandler } from "~/libs/modules/controller/controller.js";
 import { HTTPCode, HTTPError } from "~/libs/modules/http/http.js";
 
 const checkUserPermissions = (routePermissions: string[]): APIPreHandler => {
-	return (request: FastifyRequest): void => {
+	return (request: FastifyRequest): Promise<void> => {
```

Comment:
I think there is no need for this promise wrapper. As I remember we should receive `(request, response, done)` as parameters and we just need to call done at the end